### ISSUE TITLE
Shouldn't PNG and JPEG be set to True?

### ISSUE THREAD

**Issue by escorciav** [conda-forge/opencv-feedstock#70]
Hi guys,

Thank you for providing the recipe and binaries with most of OpenCV utilities. I understand the headache of compiling OpenCV and I really appreciate ur effort automating this.

I was testing it and it got the following error trying to load a video (indeed it's the same in your FFmpeg unit-test): `Unable to stop the stream: Inappropriate ioctl for device`.
I solved the issue thanks to [SO](http://stackoverflow.com/questions/42562876/opencv3-error-unable-to-stop-the-stream-inappropriate-ioctl-for-device) just setting the flags of PNG, JPEG to 1 during CMake. I do not know why the problem goes away, in any case, I am curious about the reason to set this flags to 0.

conda version: 4.3.16
OS: Linux 4.9.14-200.fc25.x86_64

code use for testing
```python
import cv2

vid = cv2.VideoCapture('VID00003-20100701-2204.avi')
assert vid.isOpened()

flag, img = vid.read()
assert flag
print(img.shape)
cv2.imwrite("whatever.jpg", img)
```
Output
```
Unable to stop the stream: Inappropriate ioctl for device
Traceback (most recent call last):
  File "cev.py", line 4, in <module>
    assert vid.isOpened()
AssertionError
```

conda environment used for testing:
```
# packages in environment at /home/escorciav/install/bin/miniconda3/envs/opencv-test-2:
#
blas                      1.1                    openblas    conda-forge
bzip2                     1.0.6                         1    conda-forge
ca-certificates           2017.1.23                     1    conda-forge
cairo                     1.14.6                        4    conda-forge
certifi                   2017.4.17                py35_0    conda-forge
ffmpeg                    3.2.4                         0    conda-forge
fontconfig                2.12.1                        4    conda-forge
freetype                  2.7                           1    conda-forge
gettext                   0.19.7                        1    conda-forge
giflib                    5.1.4                         0    conda-forge
glib                      2.51.4                        0    conda-forge
harfbuzz                  1.3.4                         2    conda-forge
hdf5                      1.8.17                       10    conda-forge
icu                       58.1                          1    conda-forge
jasper                    1.900.1                       4    conda-forge
jpeg                      9b                            0    conda-forge
libffi                    3.2.1                         3    conda-forge
libgfortran               3.0.0                         1  
libiconv                  1.14                          4    conda-forge
libpng                    1.6.28                        0    conda-forge
libtiff                   4.0.6                         7    conda-forge
libwebp                   0.5.2                         7    conda-forge
libxml2                   2.9.4                         4    conda-forge
ncurses                   5.9                          10    conda-forge
numpy                     1.12.1          py35_blas_openblas_200  [blas_openblas]  conda-forge
openblas                  0.2.19                        2    conda-forge
opencv                    3.2.0           np112py35_blas_openblas_201  [blas_openblas]  conda-forge
openssl                   1.0.2k                        0    conda-forge
pcre                      8.39                          0    conda-forge
pip                       9.0.1                    py35_0    conda-forge
pixman                    0.34.0                        0    conda-forge
python                    3.5.3                         1    conda-forge
readline                  6.2                           0    conda-forge
setuptools                33.1.1                   py35_0    conda-forge
sqlite                    3.13.0                        1    conda-forge
tk                        8.5.19                        1    conda-forge
wheel                     0.29.0                   py35_0    conda-forge
x264                      20131217                      3    conda-forge
xz                        5.2.2                         0    conda-forge
zlib                      1.2.11                        0    conda-forge
```

conda environment YAML
```
name: opencv-test-2
channels:
- conda-forge
- defaults
dependencies:
- blas=1.1=openblas
- bzip2=1.0.6=1
- ca-certificates=2017.1.23=1
- cairo=1.14.6=4
- certifi=2017.4.17=py35_0
- ffmpeg=3.2.4=0
- fontconfig=2.12.1=4
- freetype=2.7=1
- gettext=0.19.7=1
- giflib=5.1.4=0
- glib=2.51.4=0
- harfbuzz=1.3.4=2
- hdf5=1.8.17=10
- icu=58.1=1
- jasper=1.900.1=4
- jpeg=9b=0
- libffi=3.2.1=3
- libiconv=1.14=4
- libpng=1.6.28=0
- libtiff=4.0.6=7
- libwebp=0.5.2=7
- libxml2=2.9.4=4
- ncurses=5.9=10
- numpy=1.12.1=py35_blas_openblas_200
- openblas=0.2.19=2
- opencv=3.2.0=np112py35_blas_openblas_201
- openssl=1.0.2k=0
- pcre=8.39=0
- pip=9.0.1=py35_0
- pixman=0.34.0=0
- python=3.5.3=1
- readline=6.2=0
- setuptools=33.1.1=py35_0
- sqlite=3.13.0=1
- tk=8.5.19=1
- wheel=0.29.0=py35_0
- x264=20131217=3
- xz=5.2.2=0
- zlib=1.2.11=0
- libgfortran=3.0.0=1
```

**Comment by jakirkham** [comment]
Thanks for the report.

Could you explain a bit more about your setup? Is this a headless machine? Or does it have some kind of display? Is X11 in play? Really anything relating to this that might be helpful.

IIRC the `BUILD_JPEG` and `BUILD_PNG` flags specify whether to use OpenCV to build its own internal copy of these libraries or not. We tell it not to and provide packages from `conda-forge` to satisfy these dependencies. We also specify where to look for them. That said, this wouldn't be the first time OpenCV got confused and did something silly because of how we provided a flag (looking at you OpenCV Python support). In short, I don't think enabling these is the right fix, but there may be something about what OpenCV expects from these dependencies that needs to be corrected. Would be good to figure that out if this is in fact the issue.

> I was testing it and it got the following error trying to load a video (indeed it's the same in your FFmpeg unit-test): Unable to stop the stream: Inappropriate ioctl for device.

Sorry, I think I missed something, which test and where? 😕

**Comment by jakirkham** [comment]
FWIW it seems like you are not alone. ( https://github.com/opencv/opencv/issues/8097 )

**Comment by jakirkham** [comment]
Looks like we aren't picking up `libavresample` for some reason. Potentially part of the issue.

```
-- Checking for module 'libavresample'
--   No package 'libavresample' found
```

ref: https://circleci.com/gh/conda-forge/opencv-feedstock/157

**Comment by escorciav** [comment]
Thanks 4 ur reply.

1. I have a display but I was doing everything via tty. My desktop is x11.
2. Regarding the PNG JPEG flag, you are right. My fault 😓 , I tried the SO solution and I thought that I was setting `WITH_[PNG/JPEG]` as I usually do. I agree with u that they should be False. It's disappointing that they solved the issue without being connected to it in theory.
3. For the testing, I used a simple program in the first block of code in my comment.
4. It seems that `libavresample` is related with the issue. I noticed that it wasn't found when I built. However, I am not that expert on all the explicit 3rdparty libraries connections inside OpenCV.

I will check the `pkg-config` folder of the environment. Do you know if the stdout is saved by conda?

P.D. Feel free to change the title of the issue for something more appropriate e.g. the error message 😆

**Comment by escorciav** [comment]
Quick update: I compiled openCV from source using all the dependencies provided by opencv-feedstock recipe (I tried my best). According to the cmake output `libavresample` is not found **but** I can open the video and run the code without problem.

```
-- Checking for modules 'libavcodec;libavformat;libavutil;libswscale'
--   Found libavcodec, version 57.64.101
--   Found libavformat, version 57.56.101
--   Found libavutil, version 55.34.101
--   Found libswscale, version 4.2.100
-- Checking for module 'libavresample'
```

Lemme know how to remove the opencv that I built with conda-build. I did not build ur recipe as it is. I guess it can help us to find the issue.

**Comment by patricksnape** [comment]
This should be fixed now since that test passed in #75. So please confirm it's fixed (in about 2 hours) and then we can close this

**Comment by escorciav** [comment]
tested on linux-x64 and OSX and worked like a charm.
👍 , I'm glad that you found the solution to make it work with conda. It simplifies deployment.
